src/metatensor/models/soap_bpnn/train.py

```diff
@@ -44,4 +64,6 @@ def train(model, train_dataset, hypers):
             optimizer.step()
 
     # Save the model:
-    torch.save(model.state_dict(), "model_final.pt")
+    save_model(
+        ARCHITECTURE_NAME, model, model_hypers, model.all_species, "model_final.pt"
```

Comment:
can't we get the `model_hypers` from the model object? As written above I find it a bit weird that the trainer needs the model hypers.

Also, I am not sure if I like that this is inside the trainer and not controlled by the `train.py`

Reply:
We could indeed save the hypers inside the model instead. Let me know if this looks like a better solution to you

Reply:
I think so. I it confusing that the trainer also needs the model hypers AND the model even though the model already should now and saves the hypers somehow.